### PR TITLE
Don't fetch chunks outside our timerange

### PR DESCRIPTION
**Pull request by tomwilkie**
Should help with #132

### CODE REVIEW

File: chunk/chunk_store.go [hunk 1]
```diff
@@ -417,14 +417,28 @@ func (c *AWSStore) Get(ctx context.Context, from, through model.Time, matchers .
...
 
-	queryChunks.Observe(float64(len(missing)))
+	filtered := make([]Chunk, 0, len(chunks))
+	for _, chunk := range chunks {
+		_, chunkFrom, chunkThrough, err := parseChunkID(chunk.ID)
+		if err != nil {
+			return nil, err
+		}
+		if chunkThrough < from || through < chunkFrom {
+			fmt.Println(chunkThrough, from, through, chunkFrom)
```
Comment (jml) [inline chunk/chunk_store.go]: Guessing this `Println` is there for debugging. Please remove or turn into a proper log call.

File: chunk/chunk_store.go [hunk 2]
```diff
@@ -417,14 +417,28 @@ func (c *AWSStore) Get(ctx context.Context, from, through model.Time, matchers .
...
-	missing, err := c.lookupChunks(userID, from, through, matchers)
+	chunks, err := c.lookupChunks(userID, from, through, matchers)
 	if err != nil {
 		return nil, err
 	}
 
-	queryChunks.Observe(float64(len(missing)))
+	filtered := make([]Chunk, 0, len(chunks))
+	for _, chunk := range chunks {
+		_, chunkFrom, chunkThrough, err := parseChunkID(chunk.ID)
```
Comment (jml) [inline chunk/chunk_store.go]: I realise the `from` & `through` terminology was present before this PR, but I think `start` & `end` would be better. Up to you whether you want to change in this PR.

Comment (tomwilkie) [inline chunk/chunk_store.go]: I actually like `from` & `through` - `end` tends to be a reserved word in many languages, and `through` implies inclusivity.

File: chunk/chunk_store.go (hunk 1 above)
Comment (tomwilkie) [inline chunk/chunk_store.go]: Whoops!

**Pull request merged**